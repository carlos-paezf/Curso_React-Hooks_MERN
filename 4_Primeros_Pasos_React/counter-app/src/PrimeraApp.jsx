import React from 'react'
import PropTypes from 'prop-types'


const PrimeraApp = ({ nombre, apellido }) => {
    const saludo = 'Hola mundo - const'
    const bool = true
    const persona = {
        nombre: 'David',
        apellido: 'Ferrer',
        edad: 21
    }

    return (
        <>
            <h1>Bienvenido { nombre } { apellido }</h1>
            <pre>{ saludo }</pre>
            <pre>{ JSON.stringify(bool) }</pre>
            <pre>{ JSON.stringify(persona) }</pre>
        </>
    )
}

PrimeraApp.propTypes = {
    nombre: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    apellido: 'nn'
}

export default PrimeraApp
