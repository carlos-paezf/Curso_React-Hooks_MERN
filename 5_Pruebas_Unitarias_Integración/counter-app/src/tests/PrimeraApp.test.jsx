import React from 'react'
import { render, screen } from '@testing-library/react'

import { shallow } from 'enzyme'
import PrimeraApp from '../PrimeraApp'


describe('Pruebas en <PrimeraApp />', () => {
    test('Debe mostrar el mensaje "Bienvenido David nn"', () => {
        const saludo = 'Bienvenido David nn'
        render(<PrimeraApp nombre="David" />)
        expect(screen.getByText(saludo)).toBeInTheDocument()
    })

    test('Debe mostrar <PrimeraApp /> correctamente', () => {
        const wrapper = shallow(<PrimeraApp nombre='David' />)
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe mostrar el apellido enviado por props', () => {
        const wrapper = shallow(<PrimeraApp nombre='David' apellido='Ferrer' />)
        const textH1 = wrapper.find('h1').text()
        expect(textH1).toBe(`Bienvenido David Ferrer`)
    })
})
