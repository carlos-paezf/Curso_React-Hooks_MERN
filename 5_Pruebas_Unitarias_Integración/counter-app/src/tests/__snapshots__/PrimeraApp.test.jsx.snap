// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Bienvenido 
    David
     
    nn
  </h1>
  <pre>
    Hola mundo - const
  </pre>
  <pre>
    true
  </pre>
  <pre>
    {"nombre":"David","apellido":"Ferrer","edad":21}
  </pre>
</Fragment>
`;

exports[`Pruebas en <PrimeraApp /> Debe mostrar el apellido enviado por props 1`] = `
<Fragment>
  <h1>
    Bienvenido 
    David
     
    nn
  </h1>
  <pre>
    Hola mundo - const
  </pre>
  <pre>
    true
  </pre>
  <pre>
    {"nombre":"David","apellido":"Ferrer","edad":21}
  </pre>
</Fragment>
`;
